src/dataHelpers.ts

```diff
@@ -0,0 +1,77 @@
+import { Content, Pillar } from "./api";
+
+interface ImageSrcParams {
+    isCommercial?: boolean;
+    isContributor?: boolean;
+}
+
+export const getKickerText = (story: Content): string => {
+    if (!story.header.kicker) {
+        return "";
+    }
+
+    switch (story.header.kicker.type) {
+        case "FreeHtmlKicker":
+            return story.header.kicker.item.properties.kickerText || "";
+        case "LiveKicker":
+            return "Live";
+        default:
+            return "";
+    }
+};
+
+export const getPillarName = (story: Content): Pillar => {
+    // All cards
+    try {
+        const pillarName = story.properties.maybeContent.metadata.pillar.name;
+        return pillarName;
+    } catch (e) {
+        return null;
+    }
+};
+
+export const getImageSrc = (
+    story: Content,
+    options: ImageSrcParams = {}
+): string => {
+    if (options.isCommercial) {
+        return story.properties.image.item.imageSrc || null;
+    }
+
+    if (options.isContributor) {
+        const leadContributor = story.properties.maybeContent.tags.tags.find(
+            (tag: any) => {
+                return tag.properties.tagType === "Contributor";
+            }
+        );
+        if (!leadContributor) {
+            return null;
+        }
+        return leadContributor.properties.contributorLargeImagePath || null;
+    }
+
+    try {
+        const imageSrc =
+            story.properties.maybeContent.trail.trailPicture.allImages[0].url;
+        return imageSrc;
+    } catch (e) {
+        return null;
+    }
+};
+
+export const getCardUrl = (story: Content): string => {
+    const baseUrl = story.properties.webUrl
+        ? story.properties.webUrl
+        : `https://www.theguardian.com${story.properties.href}`;
+
+    const brazeParam = "?##braze_utm##";
```

Comment:
Is it worth keeping this in the components themselves? I.e. at the point of use? My thinking is that it is not a core part of the URL/model so there may be cases in the future where we want the URL without the braze param. Instead, we could create a helper function such as `withBrazeParam` to wrap URLs within components?

It's not a blocker - I can see that this saves a lot of typing. Just I can see it being potentially confusing/restrictive later down the line to modify the core model in this way.

Reply:
Good point, I just didn't find any use case where we wouldn't want the cardUrl to have the parameter as it's also used for click-thru links - but easy to update with a `withBrazeParam` and be more future-proof.